buildSrc/build.gradle.kts

```diff
@@ -14,7 +14,8 @@
  */
 
 plugins {
-  `kotlin-dsl`
+  // TODO
+  `kotlin-dsl`.version("3.2.2")
```

Comment:
This is a hack to avoid upgrading to Gradle 8.


If I'm upgrade to Gradle 8 on main without Kotlin 1.8 I get the following error:
```
A problem occurred configuring project ':tangle-fragment-api'.
> Failed to notify project evaluation listener.
   > Could not create task ':tangle-fragment-api:mergeDebugShaders'.
      > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
   > Invalid MavenPublish Configuration. Unable to find variant to publish named release. Try setting the 'androidVariantToPublish' property in the mavenPublish extension object to something that matches the variant that ought to be published.
```
Upgrading the maven publish plugin doesn't fix this either.

If I try to upgrade to Kotlin 1.8 without this above hack or Gradle 8 I get a ton of `Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.` errors for the `compileKotlin` task for the buildSrc _I'm assuming_ this is because the buildSrc is tied directly to Kotlin 1.6.

I'm going to continue to tinker around with this to see if I can find a better solution.

Reply:
This has been resolved by 9caf86fea1340e3cec831fe90e7665e4edd841b7.

I'm not sure why the ordering matter but after the initial work to upgrade to Kotlin 1.8  when I tried upgrading to Gradle 8 and the latest vanniktech publish plugin I can different error messages that were easier to resolve.